graylog2-server/src/main/java/org/graylog2/messageprocessors/MessageProcessorsConfigWithDescriptors.java

```diff
@@ -0,0 +1,84 @@
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.messageprocessors;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+import org.graylog2.plugin.messageprocessors.MessageProcessor;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@JsonAutoDetect
+@AutoValue
+public abstract class MessageProcessorsConfigWithDescriptors {
+    @JsonProperty("processor_order")
+    public abstract List<MessageProcessorDescriptor> processorOrder();
+
+    @JsonProperty("disabled_processors")
+    public abstract Set<String> disabledProcessors();
+
+    @JsonCreator
+    public static MessageProcessorsConfigWithDescriptors create(@JsonProperty("processor_order") List<MessageProcessorDescriptor> processorOrder,
+                                                                @JsonProperty("disabled_processors") Set<String> disabledProcessors) {
+        return builder()
+                .processorOrder(processorOrder)
+                .disabledProcessors(disabledProcessors)
+                .build();
+    }
+
+    public static MessageProcessorsConfigWithDescriptors fromConfig(MessageProcessorsConfig config,
+                                                                    Set<MessageProcessor.Descriptor> descriptors) {
+        final Map<String, MessageProcessor.Descriptor> descriptorMap = descriptors.stream()
+                .collect(Collectors.toMap(MessageProcessor.Descriptor::className, descriptor -> descriptor));
+
+        return builder()
+                .processorOrder(config.processorOrder().stream()
+                        .map(s -> MessageProcessorDescriptor.fromDescriptor(descriptorMap.get(s)))
+                        .collect(Collectors.toList()))
+                .disabledProcessors(config.disabledProcessors())
+                .build();
+    }
+
+    public MessageProcessorsConfig toConfig() {
+        return MessageProcessorsConfig.builder()
+                .processorOrder(processorOrder().stream()
+                        .map(MessageProcessorDescriptor::className)
+                        .collect(Collectors.toList()))
+                .disabledProcessors(disabledProcessors())
+                .build();
+    }
+
+    public static Builder builder() {
+        return new AutoValue_MessageProcessorsConfigWithDescriptors.Builder();
+    }
+
+    public abstract Builder toBuilder();
+
+    @AutoValue.Builder
+    public static abstract class Builder {
```

Comment:
They are not necessary, but I like to have AutoValue objects with builders because once you want to modify those objects it gets painful with only a `create` method. (see `withProcessors`)

In this case they are not really necessary, but I would like to keep them. :smiley: